### ISSUE TITLE
Docker support (contribution)

### ISSUE THREAD

**Issue by chennin** [eliasdoehne/stellaris-dashboard#88]
Hello! I have been running the dashboard in a Docker container (primarily because Python and Ubuntu upgrades have bit me in the past, but also to learn Dockerfiles and Github actions) and I'd like to contribute that work.

So my first question is - do you want the work here in this repository? Or would you want me to publish the Docker images under my name?

Importing in to this repository shouldn't require *much* work, perhaps some path changes. You'd also need to have a Docker Hub account and generate some secrets and save them in the repository secrets section.

(I acknowledge that the demand for this is probably quite low! But I still want to open-source the work somewhere)

What I have working (in private repositories temporarily):

1. Dockerfiles for Debian and Alpine (I may or may not drop Alpine because compiling scipy/numpy/matplotlib takes a long time and not much space is saved) and the ability to use various Python versions (depends on the [official Python docker containers](https://hub.docker.com/_/python)
2. Github actions that build the images and push to Docker Hub with tags including master, latest, v4.3, etc. (Not quite automated on pushes to *this* repository, but on a schedule and able to be manually triggered)
3. Documentation on how to set it up and use it

**Comment by eliasdoehne** [comment]
Sure, while I don't use docker much myself, I had some people request this before, so you're definitely not the only one and I think it would be a great idea to add a docker file. 

However, before setting up the workflow here I'll have to do some research on how to do this safely in a public repo since I don't have much experience with this and I don't want to make some mistake and accidentally leak the credentials in the workflow. 

Of course you're also welcome to publish the docker image on your own account with a separate if you prefer this option.

**Comment by chennin** [comment]
Ok, I've made the repo public if you want to see how I'm doing it, e.g.: https://github.com/chennin/stellaris-dashboard-docker/blob/main/.github/workflows/docker-publish.yml .  

I have somewhat overengineered it :) the other two files in that directory pass information to that file which is then used to tag and build various OS/Python combinations as well as designate one "latest"/stable. Will be good for testing new Python versions.

Secrets are well-handed by Github; you set them in the repository's settings then you refer to them in your Github Actions like this:

https://github.com/chennin/stellaris-dashboard-docker/blob/main/.github/workflows/docker-publish.yml#L68-74

It would also be cleaner to run the build actions in this repo because Github actions tend to assume that, and I've had to use more code to get the desired output. For example triggering a build when a new release is tagged is much easier.; instead I am currently using a whole action just to find the newest release tag, and a timer and manual button-pushes to build. which is probably good enough.

A couple of other todos I want to do before making an image public - change the image name, and have a new release tagged of the dashboard so Py3.11 would work and I can make the alpine Dockerfile less hacky. (No rush!)
